pkg/exporter/sumologicexporter/sender.go

```diff
@@ -609,44 +624,35 @@ func (s *sender) sendOTLPMetrics(ctx context.Context, md pmetric.Metrics) error
 // appendAndSend appends line to the request body that will be sent and sends
 // the accumulated data if the internal logBuffer has been filled (with maxBufferSize elements).
 // It returns appendResponse
-func (s *sender) appendAndSend(
+func (s *sender) appendAndMaybeSend(
 	ctx context.Context,
-	line string,
+	lines []string,
```

Comment:
This is going to be an issue if `lines` are going to be a lot of metrics (way more than reasonable value). Not sure if it's real case, but in theory it's possible

Reply:
It can be at most as many as we can have in a single Resource, and we have the whole batch in memory anyway. So at most we use 2x the memory in the most degenerate case.

Reply:
I think that's something that needs to be configured in the batch processor. `otlpexporter`, for example, doesn't expose any settings like this, it simply sends what it gets as-is, and that's what we want to use in the future. You can't send more records that are present in the batch here, and your configuration should be able to deal with this either way.

Reply:
It makes sense. Also for fluentd the batches were stored in separate files.